lib/game.rb

```diff
@@ -0,0 +1,117 @@
+class Game
+
+  attr_accessor :board, :player_1, :player_2
+
+  WIN_COMBINATIONS = [
+    [0,1,2],
+    [3,4,5],
+    [6,7,8],
+    [0,3,6],
+    [1,4,7],
+    [2,5,8],
+    [0,4,8],
+    [6,4,2]
+  ]
+
+
+  def initialize(player_1 = Human.new('X'), player_2 = Human.new('O'), board=Board.new)
+    @player_1 = player_1
+    @player_2 = player_2
+    @board = board
+    puts "     OK, let's play some Tic Tac Toe!"
+    puts "     Player 1 is X and Player 2 is O. Player 1 goes first."
+    puts "     Choose a number 1 - 9 (left to right, top to bottom)"
+    puts " "
+  end
+
+  def turn_count
+    board.count{|space| space == "X" || space == "O"}
+  end
+
+  def current_player
+   if board.turn_count % 2 == 0
+     return @player_1
+     else
+     return @player_2
+   end
+  end
+
+  def position_taken?(position)
+    if board.cells[position] == " " || board.cells[position] == "" || board.cells[position] == nil
+      false
+    else
+      true
+    end
+  end
+
+  def won?
+  WIN_COMBINATIONS.find do |wins|
+    board.cells[wins[0]] == board.cells[wins[1]] &&board.cells[wins[1]] == @board.cells[wins[2]] && position_taken?(wins[0])
+  end
+end
+
+  def full?
+    board.cells.all? do |letter|
+      letter == "X" || letter == "O"
+    end
+  end
+
+  def draw?
+    full? && !won?
+  end
+
+  def over?
+    won? || draw? || full?
```

Comment:
Are all three of these terms necessary? Can the board ever be `full?` if no player has `won?` and the game is not a `draw?`?